sktime/forecasting/compose/_pipeline.py

```diff
@@ -1583,10 +1583,14 @@ def _get_X_for_fcX(self, X):
             return None
 
         if ixx == "complement":
-            X_for_fcX = X.drop(columns=self.columns)
-            if X_for_fcX.shape[1] < 1:
+            if X is None:
```

Comment:
Can you please exaplain at what scenario this check be True? Is it not already handled in L1573?

> if X is None or ixx is None or ixx == "None":

Reply:
Yeah right!
This is redundant going to remove it.